src/codons.py

```diff
@@ -60,15 +69,25 @@ def translate_codons(codons: list[str]) -> list[str]:
     True
 
     """
-    # FIXME: Implement the function
-    return []
 
-
-def translate_dna(dna: str) -> str:
+    amino_list = []
+    
+    for codon in codons_list:
+        if codon.upper() not in CODON_MAP.keys():
```

Comment:
You don't need to check in `CODON_MAP.keys()`. you can just check `if codon.upper() not in CODON_MAP`. It does the same thing, but the latter is slightly more efficient (for reasons we have to get to the end of the class to see).